src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -115,6 +116,22 @@ internal IEnumerable<IKeyValueAdapter> Adapters
         /// </summary>
         internal bool IsOfflineCacheConfigured { get; private set; } = false;
 
+        /// <summary>
+        /// The <see cref="ILoggerFactory"/> for creating a logger to log errors.
+        /// </summary>
+        internal ILoggerFactory LoggerFactory { get; set; }
+
+        /// <summary>
+        /// Set a logger factory to log errors.
+        /// </summary>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> for creating a logger.</param>
+        /// <returns></returns>
+        public AzureAppConfigurationOptions SetLoggerFactory(ILoggerFactory loggerFactory)
```

Comment:
We don't do any logging for startup. I'm inclined to not introduce this API at all - only support obtaining logger from DI. We can introduce this API in the future if there is a need.

Reply:
Okay, I can remove it. And if someone is not using DI, they can always call GetRefresher() and set the LoggerFactory on that refresher instance manually.